pkg/kritis/util/util.go

```diff
@@ -36,16 +60,94 @@ func GetResource(image string) *grafeas.Resource {
 	return &grafeas.Resource{Uri: GetResourceURL(image)}
 }
 
-func CreateAttestationSignature(image string, pgpSigningKey *secrets.PGPSigningSecret) (string, error) {
-	hostSig, err := container.NewAtomicContainerSig(image, map[string]string{})
+func CreateAttestation(image string, pgpSigningKey *secrets.PGPSigningSecret) (*cryptolib.Attestation, error) {
+	hostSig, err := container.NewAtomicContainerSig(image, nil)
+	if err != nil {
+		return nil, err
+	}
+	payload, err := hostSig.JSON()
+	if err != nil {
+		return nil, err
+	}
+
+	privateKey, err := extractPrivateKeyBytes(pgpSigningKey)
+	if err != nil {
+		return nil, err
+	}
+	signer, err := cryptolib.NewPgpSigner(privateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	att, err := signer.CreateAttestation([]byte(payload))
+	if err != nil {
+		return nil, err
+	}
+	return att, nil
+}
+
+func extractPrivateKeyBytes(secret *secrets.PGPSigningSecret) ([]byte, error) {
```

Comment:
Question/investigation note: is there an easier way to get private key from secret? Maybe this is something we should factor into secret package?

Reply:
Discussed with @acamadeo. There is a concern of maintenance over leaving the these crypto code in `util` package. We'll consider refactor `CreateAttestation` interface or `secrets.PGPSigningSecret` to simply code here.

Reply:
After some research, it is not possible to extract a usable private key payload from a `packet.PrivateKey`. You can extract one from an `openpgp.Entity`. 

Currently, `secrets.PGPSigningSecret` holds a `packet.PublicKey` and `packet.PrivateKey`. I will write a PR that refactors `secrets.PGPSigningSecret` to instead hold an `openpgp.Entity` which contains the public and private key.